File: activemqdemo/src/main/java/com/ggd543/demo/virtualtopic/Producer.java
package com.ggd543.demo.virtualtopic;

import org.apache.activemq.ActiveMQConnection;
import org.apache.activemq.ActiveMQConnectionFactory;
import org.apache.activemq.command.ActiveMQTopic;
import scala.tools.nsc.symtab.StdNames;

import javax.jms.*;

/**
 * Created by ggd543 on 14-1-12.
 */
public class Producer {
    public static void main(String[] args) throws JMSException {
        ActiveMQConnectionFactory factory = new ActiveMQConnectionFactory(TestVirtualTopic.URL);
        Topic topic = new ActiveMQTopic(TestVirtualTopic.getVirtualTopicName());
        ActiveMQConnection conn = (ActiveMQConnection) factory.createConnection();
        Session session = conn.createTopicSession(true, Session.AUTO_ACKNOWLEDGE);
        MessageProducer producer = session.createProducer(topic);
        producer.setDeliveryMode(DeliveryMode.PERSISTENT);
        conn.start();
        int index = 0;
        while (index++ < 5) {
            TextMessage message = session.createTextMessage(index + " message.");
            System.out.println(message);
            producer.send(message);
        }
        session.commit();
//        session.rollback();
        session.close();
        conn.close();
    }


}

File: activemqdemo/src/main/scala/com/ggd543/mq/demo/Producer.scala
package com.ggd543.mq.demo

/**
 *
 * 功能描述：
 * <p> 版权所有：优视科技
 * <p> 未经本公司许可，不得以任何方式复制或使用本程序任何部分 <p>
 *
 * @author <a href="mailto:liuyj3@ucweb.com">刘永健</a>
 * @version 1.0.0
 * @since 1.0.0 
 *        create on: 2014年01月10
 */
import javax.jms._
import org.apache.activemq.ActiveMQConnection
import org.apache.activemq.ActiveMQConnectionFactory

object Producer {
  val activeMqUrl  = ActiveMQConnection.DEFAULT_BROKER_URL
  val topicName  = "TOPIC_NAME"

  def main(args: Array[String]): Unit = {
    val connectionFactory  = new ActiveMQConnectionFactory(activeMqUrl)
    val connection  = connectionFactory.createConnection
    val session  = connection.createSession(false, Session.AUTO_ACKNOWLEDGE)
    val destination  = session.createTopic(topicName)
    val messageProducer  = session.createProducer(destination)
    messageProducer.setDeliveryMode(DeliveryMode.PERSISTENT)
    messageProducer.setTimeToLive(5000)
    connection.start
    val textMessage  = session.createTextMessage("Hello Subscriber!")
    for (i <- 0 until 3) {
      messageProducer.send(textMessage)
      println("Message sent to subscriber: '" + textMessage.getText + "'")
    }
    connection.close
  }

}
